plugin/pulpcore/plugin/download/file.py

```diff
@@ -35,11 +35,13 @@ def __init__(self, url, **kwargs):
         Read, validate, and compute digests on the `url`. This is a coroutine.
 
         This method provides the same return object type and documented in
-        :meth:`~pulpcore.plugin.download.asyncio.BaseDownloader.run`.
+        :meth:`~pulpcore.plugin.download.BaseDownloader.run`.
         """
-        with open(self._path, 'r') as f_handle:
+        async def read():
```

Comment:
I see why this needed to be changed. Unlike the aiohttp objects which are _awaitable_ the `read()` from stdlib is not. I think we should consider using an aio based file like the one this library provides: https://pypi.python.org/pypi/aiofiles/0.3.2

Reply:
A little concerned aiofile is only at version 0.3 but probably a good idea.